istio/src/main/scala/io/buoyant/k8s/istio/mixer/MixerApiRequests.scala

```diff
@@ -0,0 +1,121 @@
+package io.buoyant.k8s.istio.mixer
+
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.google.local.DurationProto.{Duration => GDuration}
+import com.twitter.util.Duration
+import io.buoyant.k8s.istio._
+import istio.mixer.v1._
+
+import scala.io.Source
+
+object MixerApiRequests {
+
+  private[this] lazy val istioGlobalDict: Seq[String] = {
+    val yaml = Source.fromInputStream(
+      getClass.getClassLoader.getResourceAsStream("mixer/v1/global_dictionary.yaml")
+    ).mkString
+
+    val mapper = new ObjectMapper(new YAMLFactory) with ScalaObjectMapper
+    mapper.registerModule(DefaultScalaModule)
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+    mapper.readValue[Seq[String]](yaml)
+  }
+
+  def mkReportRequest(
+    responseCode: ResponseCodeIstioAttribute,
+    requestPath: RequestPathIstioAttribute,
+    targetService: TargetServiceIstioAttribute,
+    sourceLabel: SourceLabelIstioAttribute,
+    targetLabel: TargetLabelsIstioAttribute,
+    duration: ResponseDurationIstioAttribute
+  ): ReportRequest = {
+
+    val dictionary = mkDictionary(requestPath, targetService, sourceLabel, targetLabel)
+
+    val updated = mkAttributesUpdate(dictionary, Seq(responseCode, requestPath, targetService, sourceLabel, targetLabel, duration))
+    ReportRequest(attributeUpdate = Some(updated))
+  }
+
+  def mkCheckRequest(istioREquest: IstioRequest[_]) = {
```

Comment:
is there a reason for the capitalization in `istioREquest`?

Reply:
Also, TIOLI, but it seems to me that a lot of this function might be better represented as a series of functions on `IstioRequest`?

Reply:
Yup, the reason is that I am still not used to the new mac's keyboard and keep fat fingering everything -_-

Fixed it, thanks :)

On the second comment, I think that most of this stuff is only relevant in a `Mixer API` context, and I would not like to create any coupling between the request object and Mixer. What did you have in mind?